### PR TITLE
fix(datadog) ensure log collection is enabled + feat(infra.ci.jenkins.io) enable datadog traces collection

### PR DESCRIPTION
**Pull request by dduportal**
Follow up of  https://github.com/jenkins-infra/helpdesk/issues/2804, this PR:

- Fix a typo in the datadog chart's value (all cluster) to ensure that logs collection is enabled on the datadog agents
- Enable traces collection in infra.ci.jenkins.io through datadog agent

(edited) N.B:

Log collection in the Jenkins datadog plugin was not enabled because it blocks all controller threads if logs cannot be collected.

Tried with https://plugins.jenkins.io/datadog/#plugin-content-log-collection-for-agents mentions a custom log source file to add to the datadog agent. That one might not be present in our current setup resulting in error messages on infra.ci.jenkins.io until we find the correct setup (by switching to official datadog image instead of ours)

=> we might want to fine tune this in the future if we want log collection (or not using datadog plugin for collecting build logs).

### CODE REVIEW

**Comment by dduportal** [comment]
Waiting for weekly release to be published before merging and deploying

**Review by timja — APPROVED** [review]

**Comment by dduportal** [comment]
EDIT: disabled the datadog logs collection by removing the 3rd commit with the change below, because datadog plugin was blocking infra.ci.jenkins.io threads and no builds were treated.

```diff
diff --git a/config/ext_jenkins-infra.yaml b/config/ext_jenkins-infra.yaml
index bd21bc5..dd2396e 100644
--- a/config/ext_jenkins-infra.yaml
+++ b/config/ext_jenkins-infra.yaml
@@ -396,7 +396,7 @@ controller:
         unclassified:
           datadogGlobalConfiguration:
             ciInstanceName: "infra.ci.jenkins.io"
-            collectBuildLogs: false
+            collectBuildLogs: true
             # Send events on configuration changes to jobs or changes to jenkins. These events include changes by the system which may be frequent and redundant.
             emitConfigChangeEvents: true
             emitSecurityEvents: true
@@ -411,6 +411,7 @@ controller:
             retryLogs: true
             targetApiURL: "https://api.datadoghq.com/api/"
             targetHost: "datadog.datadog.svc.cluster.local"
+            targetLogCollectionPort: 10518
             targetLogIntakeURL: "https://http-intake.logs.datadoghq.com/v1/input/"
             targetPort: 8125
             targetTraceCollectionPort: 8126
```

**Pull request merged**
